File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.cc
#include <deque>
#include <iostream>
#include <vector>
using namespace std;

int main(void) {
    int n, k, st;
    cin >> n >> k;
    deque<int> dq;
    vector<int> res;
    st = k;
    for (int i = 0; i < n; i++)
        dq.push_back(i + 1);
    while (res.size() < n) {
        res.push_back(dq[st - 1]);
        dq.erase(dq.begin() + st - 1);
        if (dq.size() > 0) {
            st = (st - 1 + k) % dq.size();
            if (st == 0) st = dq.size();
        }
    }
    cout << "<";
    for (int i = 0; i < res.size(); i++)
        cout << res[i] << (i == res.size() - 1 ? ">" : ", ");
    return 0;
}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys
from collections import deque
n,k=map(int,sys.stdin.readline().split())
dq=deque([i for i in range(1,n+1)])
st,res=k,[]
while len(res)<n:
    res+=[dq[st-1]]
    del dq[st-1]
    if len(dq)>0:
        st=(st-1+k)%len(dq)
        if st==0: st=len(dq)
print("<",end='')
for i in res:
    print(i,end='' if i==res[-1] else ", ")
print(">")